src/common/security/robot/robot.go

```diff
@@ -60,3 +56,20 @@ func filterPolicies(namespace rbac.Namespace, policies []*rbac.Policy) []*rbac.P
 
 	return results
 }
+
+// getAllPolicies gets all of supported policies supported in project and external policies supported for robot account
+func getAllPolicies(namespace rbac.Namespace) map[string]bool {
+	mp := map[string]bool{}
+	for _, policy := range project.GetAllPolicies(namespace) {
+		mp[policy.String()] = true
+	}
+
+	scannerPull := &rbac.Policy{Resource: rbac.ResourceRepository, Action: rbac.ActionCreate}
+	robotExternalPolicies := []*rbac.Policy{}
+	robotExternalPolicies = append(robotExternalPolicies, scannerPull)
+	for _, policy := range robotExternalPolicies {
+		mp[policy.String()] = true
+	}
```

Comment:
1. loop is not need here
2. action is not correct in line 67 it should be `rbac.ActionScannerPull`

Reply:
1, in case we have other exceptions. leave the array here?
2, updated, thanks.